openwisp_controller/mixins.py

```diff
@@ -35,3 +38,14 @@ class RelatedDeviceProtectedAPIMixin(
 
 class ProtectedAPIMixin(BaseProtectedAPIMixin, FilterByOrganizationManaged):
     pass
+
+
+class AutoRevisionMixin(RevisionMixin):
+    def dispatch(self, request, *args, **kwargs):
+        if request.method in ('GET', 'HEAD', 'OPTIONS'):
+            return super().dispatch(request, *args, **kwargs)
+        with reversion.create_revision():
+            response = super().dispatch(request, *args, **kwargs)
+            reversion.set_user(request.user)
+            reversion.set_comment(f'API request: {request.method} {request.path}')
+            return response
```

Comment:
@dee077 doesn't RevisionMixin already handles this? 

https://github.com/etianen/django-reversion/blob/9a7cd7419121e56d65247b050b7aa8a105aa000c/reversion/views.py#L36-L60

Apart from adding a comment that this change was made from the REST API, we are not doing any extra. In my opinion, the comment does not add much value.  While reviewing the list of revisions (at the time of restoring revisions). I would prefer simplifying the logic here and use the default behavior.

Reply:
@pandafy I think knowing the change was made via the API is useful for our commercial support service, if users are doing changes automatically via the REST API and something breaks due to those changes we can find out with less time, so I think this is a plus worth keeping.